src/db-connection.ts

```diff
@@ -0,0 +1,77 @@
+import { config } from 'node-config-ts'
+import type { Knex } from 'knex'
+import knex from 'knex'
+import snakeCase from 'lodash.snakecase'
+import camelCase from 'lodash.camelcase'
+import { logger } from './logger/index.js'
+
+const KNEX_TABLES = ['knex_migrations', 'knex_migrations_lock']
+
+function toCamelCase(value) {
+  if (Array.isArray(value)) {
+    return value.map(toCamelCase)
+  }
+
+  if (typeof value === 'object') {
+    return Object.fromEntries(Object.entries(value).map(([key, value]) => [camelCase(key), value]))
+  }
+
+  if (typeof value === 'string') {
+    return camelCase(value)
+  }
+
+  return value
+}
+
+async function runMigrations(connection: Knex) {
+  const [_, pendingMigrations] = await connection.migrate.list()
```

Comment:
do you know how knex decides which migrations have already been run or not?

Reply:
knex has a migration table that keeps track of this